backend/src/apiserver/storage/pipeline_store_kubernetes.go

```diff
@@ -0,0 +1,520 @@
+		return pipelines, len(pipelines), "", nil
+	} else if upperBound >= len(pipelines) {
+		upperBound = len(pipelines)
+		npt, err = opts.NextPageToken(pipelines[upperBound-1])
+	} else {
+		npt, err = opts.NextPageToken(pipelines[upperBound])
+	}
+	return pipelines[nextPage:upperBound], opts.PageSize, npt, err
+}
+
+func (k *PipelineStoreKubernetes) GetPipeline(pipelineId string) (*model.Pipeline, error) {
+	k8sPipeline, err := k.getK8sPipeline(pipelineId)
+	if err != nil {
+		return nil, err
+	}
+
+	return k8sPipeline.ToModel(), nil
+}
+
+func (k *PipelineStoreKubernetes) GetPipelineWithStatus(pipelineId string, status model.PipelineStatus) (*model.Pipeline, error) {
+	pipeline, err := k.GetPipeline(pipelineId)
+	if err != nil {
+		return nil, err
+	}
+
+	if pipeline.Status != status {
+		return nil, util.NewResourceNotFoundError("Pipeline", pipelineId)
+	}
+
+	return pipeline, nil
+}
+
+func (k *PipelineStoreKubernetes) DeletePipeline(pipelineId string) error {
+	k8sPipeline, err := k.getK8sPipeline(pipelineId)
+	if err != nil {
+		if strings.Contains(err.Error(), "ResourceNotFoundError:") {
+			return nil
+		}
+	}
+
+	err = k.client.Delete(k.context, k8sPipeline)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return util.NewInternalServerError(err, "Failed to delete the pipeline")
+	}
+
+	return nil
+}
+
+func (k *PipelineStoreKubernetes) CreatePipelineAndPipelineVersion(pipeline *model.Pipeline, pipelineVersion *model.PipelineVersion) (*model.Pipeline, *model.PipelineVersion, error) {
+	pipeline.UUID = ""
+	pipelineVersion.UUID = ""
+
+	var err error
+
+	pipeline, err = k.CreatePipeline(pipeline)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pipelineVersion, err = k.createPipelineVersionWithPipeline(pipeline, pipelineVersion)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return pipeline, pipelineVersion, nil
+}
+
+func (k *PipelineStoreKubernetes) CreatePipeline(pipeline *model.Pipeline) (*model.Pipeline, error) {
+	pipeline.UUID = ""
+
+	if pipeline.Parameters != "" {
+		return nil, util.NewBadRequestError(ErrUnsupportedField, "The parameters field is not supported")
+	}
+
+	if pipeline.Namespace == "" {
+		if common.IsMultiUserMode() || common.GetPodNamespace() == "" {
+			return nil, util.NewBadRequestError(errors.New("A namespace is required"), "")
+		}
+
+		pipeline.Namespace = common.GetPodNamespace()
+	}
+
+	pipeline.Status = model.PipelineCreating
+
+	k8sPipeline := v2beta1.FromPipelineModel(*pipeline)
+
+	glog.Infof("Creating the pipeline %s/%s in Kubernetes", k8sPipeline.Namespace, k8sPipeline.Name)
+
+	err := k.client.Create(k.context, &k8sPipeline)
+	if k8serrors.IsAlreadyExists(err) {
+		return nil, util.NewAlreadyExistError(
+			"Failed to create a new pipeline. The name %v already exists. Please specify a new name", pipeline.Name,
+		)
+	} else if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to create the pipeline")
+	}
+
+	err = k.client.Status().Update(k.context, &k8sPipeline)
```

Comment:
@mprahl Do we need this code? During integration tests, I found that test is failing with a `Not Found` message even though the pipeline was successfully created. It happens only on KinD, I tested with OpenShift and don't see that message.